src/distributions.jl

```diff
@@ -3,11 +3,11 @@
 function default_range(dist::Distribution, alpha = 0.0001)
     minval = isfinite(minimum(dist)) ? minimum(dist) : quantile(dist, alpha)
     maxval = isfinite(maximum(dist)) ? maximum(dist) : quantile(dist, 1-alpha)
-    minval, maxval
+    promote(minval, maxval)
 end
 
 function default_range(m::Distributions.MixtureModel, alpha = 0.0001)
-    minval = maxval = 0.0
+    minval = maxval = 0
```

Comment:
I guess this is more likely to be type unstable than the current implementation?

Reply:
Why? It turns out this is necessary for the zero-inflated Poisson to not make everything a `Float64`. Since `logpdf` for discrete `MixtureModel` only accepts points of type `Int`, this turning everything into floats causes failures.

Reply:
Because in the loop the type of `minval` and `maxval` will be changed for most distributions: for most distributions `minimum`, `maximum` and `quantile` do not return values of type `Int`. It's also not guaranteed that they are of type `Float64` but this is definitely more likely for continuous distributions (and it would still be type stable for distributions where these values are of type `Int` if one uses `min` and `max` to update `minval` and `maxval` in the loop).

`logpdf` and `pdf` accept `Real` regardless if the distribution is discrete or not. If there's a problem with MixtureModel, it has to be fixed in Distributions.

Reply:
> Because in the loop the type of `minval` and `maxval` will be changed for most distributions: for most distributions `minimum`, `maximum` and `quantile` do not return values of type `Int`. It's also not guaranteed that they are of type `Float64` but this is definitely more likely for continuous distributions (and it would still be type stable for distributions where these values are of type `Int` if one uses `min` and `max` to update `minval` and `maxval` in the loop).

We should support both though, not just continuous distributions. I pushed an implementation using `mapreduce` that is simpler and does this.